### PR TITLE
Hl 934 ahjo auth

### PR DESCRIPTION
**Pull request by rikuke**
## Description :sparkles:
This PR lays the foundation for further AHJO REST api development. See details and how to use [here].(https://helsinkisolutionoffice.atlassian.net/wiki/spaces/KAN/pages/8392311697/Django+-+AHJO+Rest+API+autentikaation+kuvaus) and in the Readme
## Testing :alembic:
Unit tests for the AhjoConnector: `pytest applications/tests/test_ahjo_authentication.py`.
Manual testing: check the readme.

## Screenshots :camera_flash:
Screenshot 2023-10-03 at 13 58 05
Screenshot 2023-10-06 at 12 42 46

### CODE REVIEW

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2316.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2316.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2316.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2316.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe handler result is __failed__ for https://helsinkilisa-ui-handler-pr2316.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2316.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2316.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe handler result is __failed__ for https://helsinkilisa-ui-handler-pr2316.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2316.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2316.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2316.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2316.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe handler result is __failed__ for https://helsinkilisa-ui-handler-pr2316.dev.hel.ninja :crying_cat_face::anger::boom::boom:

File: backend/benefit/applications/services/ahjo_authentication.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+            expires_in = token_data.get("expires_in", "")
+            return AhjoToken(
+                access_token=access_token,
+                refresh_token=refresh_token,
+                expires_in=datetime.fromisoformat(expires_in),
+            )
+        except ObjectDoesNotExist:
+            return None
+
+    def _check_if_token_is_expired(self, expires_in: datetime) -> bool:
```
Comment (mjturt) [inline backend/benefit/applications/services/ahjo_authentication.py:127]: Looks like static method

File: backend/benefit/applications/models.py [hunk 2]
```diff
@@ -864,3 +864,8 @@ class ReviewState(models.Model):
     benefit = models.BooleanField(default=False, verbose_name=_("benefit"))
     employment = models.BooleanField(default=False, verbose_name=_("employment"))
     approval = models.BooleanField(default=False, verbose_name=_("approval"))
+
+
+class AhjoSetting(models.Model):
+    name = models.CharField(max_length=255, unique=True)
```
Comment (mjturt) [inline backend/benefit/applications/models.py:870]: Timestamp could be useful here?

**Review by mjturt — APPROVED** [review]

**Pull request merged**